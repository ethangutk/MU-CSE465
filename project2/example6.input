(+ 3.14 (* 4 7))