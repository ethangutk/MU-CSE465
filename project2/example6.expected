==================================================
Input: (+ 3.14 (* 4 7))
--------------------------------------------------
Tokens
--------------------------------------------------
LITERAL              : (
ID                   : +
REAL                 : 3.14
LITERAL              : (
ID                   : *
INT                  : 4
INT                  : 7
LITERAL              : )
LITERAL              : )
--------------------------------------------------
Parse Tree
--------------------------------------------------
Program                                 
  SExpr                                 
    List                                
      LITERAL                            (
      Seq                               
        SExpr                           
          Atom                          
            ID                           +
        Seq                             
          SExpr                         
            Atom                        
              REAL                       3.14
          Seq                           
            SExpr                       
              List                      
                LITERAL                  (
                Seq                     
                  SExpr                 
                    Atom                
                      ID                 *
                  Seq                   
                    SExpr               
                      Atom              
                        INT              4
                    Seq                 
                      SExpr             
                        Atom            
                          INT            7
                LITERAL                  )
      LITERAL                            )
--------------------------------------------------
